### PR TITLE
a few small fix

### PR DESCRIPTION
**Pull request by smallshen**

### CODE REVIEW

**Comment by Fatalll** [comment]
Thanks for this PR! But, unfortunately, we are currently not ready to migrate to Kotlin 1.5 due to our code generation mechanism.

**Comment by smallshen** [comment]
Is it caused by this?
https://github.com/JetBrains-Research/kinference-primitives

**Comment by smallshen** [comment]
It is easy to fix, I'll fix those on Thursday or this weekend, I got too much homework to do this week

**Comment by smallshen** [comment]
I sugget use Kotlin IR to generate ir code directly or use ksp(https://github.com/google/ksp). instead of replacing "Primitive" string.

File: ndarray/src/commonMain/kotlin/io/kinference/ndarray/arrays/PrimitiveNDArray.kt [hunk 1]
```diff
@@ -100,7 +100,7 @@ open class PrimitiveNDArray(array: PrimitiveTiledArray, strides: Strides) : Numb
 
         val sum = t * (ERF_COEF[0] + t * (ERF_COEF[1] + t * (ERF_COEF[2] + t * (ERF_COEF[3] + t * ERF_COEF[4]))))
 
-        return (sign * (1.0 - sum * exp(-doubleValue * doubleValue))).toPrimitive()
+        return (sign * (1.0 - sum * exp(-doubleValue * doubleValue))).toInt().toPrimitive()
```
Comment (Fatalll) [inline ndarray/src/commonMain/kotlin/io/kinference/ndarray/arrays/PrimitiveNDArray.kt:103]: Unfortunately, you can't just cast float type to int and then to float again

**Comment by Fatalll** [comment]
> I sugget use Kotlin IR to generate ir code directly

Generated code is part of our open API, so we can't use IR.

> use ksp(https://github.com/google/ksp). instead of replacing "Primitive" string.

KSP is almost the same as our current approach, but I'm not sure if it provides the same flexibility.

**Pull request closed without merging**